.grit/patterns/js/intelligent_useEffect_to_useLayoutEffect.md

```diff
@@ -0,0 +1,124 @@
+---
+title: [React2Hooks] Intelligent useEffect vs useLayoutEffect
+---
+
+If a useEffect depends on layout etc. it should switch to useLayoutEffect.
+
+tags: #fix
+
+```grit
+engine marzano(0.1)
+language js
+
+
+or {
+ `useEffect($body)` as $effectHook where {
+    $body <: contains `document.$method`,
+    $effectHook => `useLayoutEffect($body)`
+  },
+ `React.useEffect($body)` as $effectHook where {
+  $body <: contains `document.$method`,
+  $effectHook => `React.useLayoutEffect($body)`
+ }
+}
```

Comment:
Not really idiomatic - this could be much more DRY.

```suggestion
  or {`$effectHook($body)`, `React.$effectHook($body)` } where {
    $effectHook => `useEffect` => `useLayoutEffect`,
    $body <: contains `document.$method`,
  },
```

Reply:
Hi @morgante, updated the suggested changes